src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
a string containing zeros is entirely different from a zero-length string.
The assertion would be triggered by Xor(""), which is clearly undefined.
(an empty string would be repeated infinite times to cover the buffer?)